vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimPsiService.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2003-2024 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.api
+
+import com.maddyhome.idea.vim.common.TextRange
+
+public interface VimPsiService {
```

Comment:
PSI is an intellij-specific term. As vim-engine is unbound from intellij, this would be better not to use such terms as well.
Inside of IntelliJ IDEA, use of PSI is a more implementation-specific thing. Looking at the methods of this class, it seems that it answers the questions about the language syntax, so probably `VimLanguageSyntaxService`?